src/components/deal-proposal-form/ProviderSelector.tsx

```diff
@@ -120,7 +102,42 @@ export function ProviderSelector({
       case "loaded":
       case "connected": {
         return (
-          <div className="grid gap-4">{providers.size === 0 ? <NoProviders /> : <Providers />}</div>
+          <div className="grid gap-4">
+            <Controller
+              control={control}
+              name={name}
+              render={({ field }) => (
+                <div className="min-w-md max-w-md">
+                  {providers.size === 0 ? (
+                    <NoProviders />
+                  ) : (
+                    Array.from(providers.entries()).map(([accountId, provider]) => {
+                      // not sure if/how to make sure field is string[] at the type level
```

Comment:
This should help: https://www.typescriptlang.org/docs/handbook/2/narrowing.html#typeof-type-guards